lib/configuration.js

```diff
@@ -31,87 +33,216 @@ var errors = therror.register({
 });
 
 /**
- * Configuration class inheriting from nconf.
- * Configuration files must be JSON files and support comments (hjson).
+ * Configuration class to merge configuration from:
  *
- * It overrides the following nconf functions:
- * a) env() method to rename the environment variables into camelCase format.
- * b) file() to support both file and directory. If it is a directory, then it merges all the JSON files
- *    in the directory. Note that the directory is not traversed recursively.
+ *  - file(path): configuration files (if path targets a configuration file) or configuration directories
+ *    (if path targets a directory). With configuration directories, all the configuration files stored
+ *    in the directory are merged in alphabetical order. Configuration files must be either JSON files
+ *    (supporting comments with hjson) or YML files.
  *
- * NOTE: The nconf library only admits one single instance. Instantiating Configuration multiple times will
- * address the same nconf instance.
+ *  - env(whitelist): environment variables. An optional white list can be used to restrict which environment
+ *    variables are to be merged.
  *
+ *  - argv(whitelist): command-line arguments. An optional white list can be used to restrict which arguments
+ *    are to be merged.
+ *
+ * NOTE: On contrast with nconf, this module supports multiple instances.
+ *
+ * @param {Object} opts
+ *    Optional object with configuration options. Currently supported:
+ *      - delimiter. Key delimiter (by default ':' to maintain compatibility with nconf).
  * @return {Object}
- *    Configuration instance (inheriting nconf methods).
+ *    Configuration instance.
  * @constructor
  */
-function Configuration() {
+function Configuration(opts) {
+
+  var delimiter = opts && opts.delimiter || ':';
+
+  var config = {};
+
+  /**
+   * Get the value identified by the array of keys in an object. For example, for an object:
+   *     {key1: {key11: 'value'}}
+   * with the keys: ['key1', 'key11']
+   * it returns: 'value'.
+   *
+   * @param {Object} object
+   * @param {String[]} keys
+   * @return {Object}
+   *    The value referenced by the keys.
+   */
+  function _getObject(object, keys) {
+    var key = keys.shift();
+    if (keys.length === 0) {
+      return object[key];
+    }
+    if (typeof object[key] === 'object' && object[key] !== null) {
+      return _getObject(object[key], keys);
+    }
+    return undefined;
+  }
+
+  /**
+   * Build an object with a hierarchy of keys array and assigning the value to the leaf element.
+   *
+   * The invocation:
+   *    _buildObject({}, 'value', ['key1', 'key11'])
+   * returns the object:
+   *    {key1: {key11: 'value'}}
+   *
+   * @param {Object} object
+   * @param {String[]} keys
+   * @return {Object}
+   *    The object enriched with the value at keys reference.
+   */
+  function _buildObject(object, value, keys) {
```

Comment:
I'll use both methods. However, it will make it incompatible with nconf because it used ':' instead of '.'. I think it's more natural '.'.